ci(trivy): fix scan failures due to TOOMANYREQUESTS DB error

aldousalvarez opened the pull request:
## Commit to be reviewed
---
ci(trivy): fix scan failures due to TOOMANYREQUESTS DB error
```
Primary Changes
----------------
1. Updated the ci.yaml to make the trivy scan 
   run only on the weekends
```
Fixes #3652

**Pull Request Requirements**
- [ ] Rebased onto `upstream/main` branch and squashed into single commit to help maintainers review it more efficient and to avoid spaghetti git commit graphs that obfuscate which commit did exactly what change, when and, why.
- [ ] Have git sign off at the end of commit message to avoid being marked red. You can add `-s` flag when using `git commit` command. You may refer to this [link](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) for more information.
- [ ] Follow the Commit Linting specification. You may refer to this [link](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#specification) for more information. 

**Character Limit**
- [ ] Pull Request Title and Commit Subject must not exceed 72 characters (including spaces and special characters).
- [ ] Commit Message per line must not exceed 80 characters (including spaces and special characters).

**A Must Read for Beginners**
For rebasing and squashing, here's a [must read guide](https://github.com/servo/servo/wiki/Beginner's-guide-to-rebasing-and-squashing) for beginners.

outSH commented:
Looks fine, but I'd rather have a separate "weekly" job triggered by cron (https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#schedule), it will be easier to see and add new weekly jobs, and it doesn't convolute our CI job with more prerequisites. @petermetz any thoughts?

petermetz commented:
> Looks fine, but I'd rather have a separate "weekly" job triggered by cron (https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#schedule), it will be easier to see and add new weekly jobs, and it doesn't convolute our CI job with more prerequisites. @petermetz any thoughts?

@outSH Can't say I disagree. I would love to remove the code from the diff that calculates the day of the week.

petermetz reviewed (changes requested):
@aldousalvarez Looks good in general, but please incorporate the change request from @outSH

petermetz commented on .github/workflows/ci.yaml:
```diff
@@ -2300,4 +2300,7 @@ name: Cactus_CI
\ No newline at end of file
+      - dev
+  schedule:
+    # Run at 8:00 AM UTC on weekends (Saturday and Sunday)
```
@aldousalvarez Looking better, thank you for the updates! If you want to run it twice a week I'd recommend Monday and Thursday. Statistically speaking it should have a higher probability of catching things as early as possible compared to it running twice in relatively quick succession over the weekend. @outSH Thoughts?

outSH replied:
I was thinking about adding a separate workflow that would be run beside the CI, but this is OK too :)

As for the schedule, I agree to have wider gap between the runs, Monday and Thursday sounds great

aldousalvarez replied:
@petermetz, @outSH updated it to Monday and Thursday. Re-requested for review. Thank you

petermetz reviewed (approved):
LGTM

petermetz reviewed (changes requested):
@aldousalvarez On second thought, are you sure you didn't just disable running ci.yaml in pull requests completely? I noticed after a rebase that the jobs are not running at all. Could be a coincidence but it could also be that the new scheduling overrides the other declarations that make it so that ci.yaml should run for each PR?

aldousalvarez commented:
@petermetz Okay, I will take a look into this. I think the scheduled workflow will only run on the default branch so I will test it on my fork.

petermetz commented:
> @petermetz Okay, I will take a look into this. I think the scheduled workflow will only run on the default branch so I will test it on my fork.

@aldousalvarez Thank you! As I'm typing this, the checks on this PR are still stuck in 'waiting for status to be reported' so it is looking more and more like the schedule is having some sort of negative side effect here (I'd love to be wrong here though)

```
build-dev Expected — Waiting for status to be reported
Required
cactus-common Expected — Waiting for status to be reported
Required
cactus-core Expected — Waiting for status to be reported
Required
cactus-core-api Expected — Waiting for status to be reported
Required
yarn_codegen Expected — Waiting for status to be reported
Required
yarn_custom_checks Expected — Waiting for status to be reported
Required
yarn_lint Expected — Waiting for status to be reported
Required
yarn_tools_validate_bundle_names Expected — Waiting for status to be reported
Required
```

aldousalvarez commented:
@petermetz It is now working. What I did was I cleaned the code and added the changes again, I think it is some lint error or whitespaces that is why it does not read the condition before.

petermetz reviewed (approved):
@aldousalvarez Got it, thank you!
